### ISSUE TITLE
Support literal map keys in destructuring `:or` (Clojure 1.13 alpha4)

### ISSUE THREAD

**Issue by jramosg** [clj-kondo/clj-kondo#2925]
**version**

clj-kondo master at commit `9075d6a7`.

**macro usage**

No macros.

**platform**

JVM, reproduced with the project command line.

**editor**

Not applicable.

**problem**

Clojure 1.13.0-alpha4 changed map destructuring so `:or` accepts both
binding-symbol-to-default and map-key-to-default entries.

clj-kondo still requires each key in `:or` to be a simple symbol. It therefore
rejects valid literal map keys:

```clojure
(let [{:keys [x]
       :or {:x 1}}
      {}]
  x)
;; => 1
```

Linting this form reports:

```text
error: Keys in :or should be simple symbols.
```

Clojure 1.13.0-alpha3 rejects the form, but alpha4 accepts it:

```shell
clojure \
  -Sdeps '{:deps {org.clojure/clojure {:mvn/version "1.13.0-alpha4"}}}' \
  -M -e '(let [{:keys [x] :or {:x 1}} {}] x)'
```

```text
1
```

The [Clojure 1.13.0-alpha4 release notes](https://clojure.org/news/2026/07/14/clojure-1-13-alpha4) describe this change:

> `:or` now accepts key→val mappings in addition to binding→val.

Literal defaults can also refer to keys listed after `&`:

```clojure
(let [{:keys [& :rebilling :repeat]
       referralamount :amount
       :or {referralamount 0
            :rebilling false
            :repeat false}}
      {:amount 13}]
  [referralamount])
;; => [13]
```

Entries after `&` remain literal map keys. They do not create local bindings.
For example, `rebilling` and `repeat` should remain unresolved here:

```clojure
(let [{:keys [& :rebilling :repeat]
       :or {:rebilling false
            :repeat false}}
      {}]
  [rebilling repeat])
```

**expected behavior**

clj-kondo should:

- accept literal map keys in `:or` when they refer to keys named in the
  destructuring form;
- continue accepting binding symbols in `:or`;
- accept literal defaults for keys listed after `&`;
- continue reporting names after `&` as unresolved when code uses them as local
  bindings.

This is a Clojure 1.13.0-alpha4 compatibility issue.
